internal/biz/resources/resources.go

```diff
@@ -16,55 +18,85 @@ import (
+	ErrDatabaseError                = errors.New("db error while querying for resource")
+	ErrResourceAlreadyExists        = errors.New("resource already exists")
+	ErrInvalidInventoryResourceID   = errors.New("inventory ID does not exist")
+	ErrInvalidInventoryResourceType = errors.New("invalid resource type")
 )
 
 type Usecase struct {
-	repository         ResourceRepository
-	Authz              authzapi.Authorizer
-	Eventer            eventingapi.Manager
-	Namespace          string
-	log                *log.Helper
-	Server             server.Server
-	DisablePersistence bool
+	repository                  ResourceRepository
+	inventoryResourceRepository InventoryResourceRepository
+	Authz                       authzapi.Authorizer
+	Eventer                     eventingapi.Manager
+	Namespace                   string
+	log                         *log.Helper
+	Server                      server.Server
+	DisablePersistence          bool
 }
 
-func New(repository ResourceRepository, authz authzapi.Authorizer, eventer eventingapi.Manager, namespace string, logger log.Logger, disablePersistence bool) *Usecase {
+func New(repository ResourceRepository, inventoryResourceRepository InventoryResourceRepository, authz authzapi.Authorizer, eventer eventingapi.Manager, namespace string, logger log.Logger, disablePersistence bool) *Usecase {
 	return &Usecase{
-		repository:         repository,
-		Authz:              authz,
-		Eventer:            eventer,
-		Namespace:          namespace,
-		log:                log.NewHelper(logger),
-		DisablePersistence: disablePersistence,
+		repository:                  repository,
+		inventoryResourceRepository: inventoryResourceRepository,
+		Authz:                       authz,
+		Eventer:                     eventer,
+		Namespace:                   namespace,
+		log:                         log.NewHelper(logger),
+		DisablePersistence:          disablePersistence,
 	}
 }
 
 func (uc *Usecase) Create(ctx context.Context, m *model.Resource) (*model.Resource, error) {
 	ret := m // Default to returning the input model in case persistence is disabled
+	updatedResources := []*model.Resource{}
 
 	if !uc.DisablePersistence {
 		// check if the resource already exists
-		resource, err := uc.repository.FindByReporterResourceId(ctx, model.ReporterResourceIdFromResource(m))
+		existingResource, err := uc.repository.FindByReporterData(ctx, m.ReporterId, m.ReporterResourceId)
+		if err != nil && errors.Is(err, gorm.ErrRecordNotFound) {
+			// Deprecated: fallback case for backwards compatibility
+			existingResource, err = uc.repository.FindByReporterResourceId(ctx, model.ReporterResourceIdFromResource(m))
+		}
+
 		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
 			return nil, ErrDatabaseError
 		}
-		if resource != nil {
+
+		if existingResource != nil {
 			return nil, ErrResourceAlreadyExists
 		}
 
-		ret, err = uc.repository.Save(ctx, m)
+		// Inventory Resource Validations
+		if m.InventoryId != nil {
```

Comment:
This is where I was concerned about throwing an exception if we do not find a record for a given InventoryResourceId, but now I understand that flow will never enter this block since we are setting InventoryId  to uuid.Nil earlier in the flow.